src/ui/hooks.js

```diff
@@ -38,21 +39,33 @@ export function useDebouncedCallback(callback, timeLimit = 250) {
     () => () => {
       timeout.current && clearTimeout(timeout.current);
       timeout.current = null;
+      lastPendingArgs.current && storedCallback.current(...lastPendingArgs.current);
+      lastPendingArgs.current = null;
     },
     [timeLimit, storedCallback]
   );
 
-  return useCallback((...args) => {
-    const oldCurrent = timeout.current;
-    if (oldCurrent) {
-      clearTimeout(oldCurrent);
-    }
+  const debouncedCallback = useCallback((...args) => {
+    lastPendingArgs.current = args;
 
+    timeout.current && clearTimeout(timeout.current);
     timeout.current = setTimeout(() => {
       timeout.current = null;
-      storedCallback.current(...args);
+      lastPendingArgs.current && storedCallback.current(...lastPendingArgs.current);
+      lastPendingArgs.current = null;
     }, timeLimit);
   }, [timeLimit, storedCallback]);
+
+  if (!debouncedCallback.flush) {
+    debouncedCallback.flush = function flush() {
+      timeout.current && clearTimeout(timeout.current);
+      timeout.current = null;
+      lastPendingArgs.current && storedCallback.current(...lastPendingArgs.current);
+      lastPendingArgs.current = null;
```

Comment:
To make the code a bit easier to follow and reduce this repetition in all 3 cases, maybe you could extract lines 63-64  into a `executePendingCallback` method. I think the name would help.

Reply:
Good idea, extract them (and also the two lines above) into separate functions.

Reply:
👍